dev/NavigationView/NavigationView.cpp

```diff
@@ -2139,11 +2143,40 @@ void NavigationView::OnNavigationViewItemTapped(const winrt::IInspectable& sende
     }
 }
 
+void NavigationView::OnNavigationViewItemKeyUp(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    // Since we handle space and enter upon initial key down, user can hold down and items get invoked rapidly
+    // To prevent that, we detect whether a key was released or not.
+    if ((args.OriginalKey() == winrt::VirtualKey::GamepadA
+        || args.Key() == winrt::VirtualKey::Enter
+        || args.Key() == winrt::VirtualKey::Space))
+    {
+        m_keyWasReleased = true;
+    }
+}
+
 void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
 {
-    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    if ((args.OriginalKey() == winrt::VirtualKey::GamepadA
+        || args.Key() == winrt::VirtualKey::Enter
+        || args.Key() == winrt::VirtualKey::Space))
     {
-        HandleKeyEventForNavigationViewItem(nvi, args);
+        // Only handle those keys if m_keyWasReleased is true!
+        if (m_keyWasReleased)
```

Comment:
>m_keyWasReleased [](start = 12, length = 16)

I think this variable needs to be more specific, something like m_invokeKeyWasReleased.  Also, what happens if you hold down space and hit enter?

Reply:
Update variable name.

If you hold space and then hit enter for the first time, the event does not get raised. If you hit enter a second time, the item gets invoked, the same for the third and fourth ... time.

Reply:
@ojhad do you think this would be an issue?

Reply:
It's not ideal but I don't see this being an issue. Seems like an edge case where, if it does happen, the workaround is just to hit enter again. And then letting go of the space bar shouldn't leave it in a bad state since its just setting `m_invokeKeyWasReleased` to true which unblocks.